x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/MvExpand.java

```diff
@@ -51,6 +58,7 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeNamedWriteable(child());
         out.writeNamedWriteable(target());
         out.writeNamedWriteable(expanded());
+        assert limit == null;
```

Comment:
I guess this never triggers b/c there's currently always going to be a limit below MV_EXPAND, so it'll never be serialized in the fragment, right?

Reply:
It never happens because we do the serialization after the physical planning, and the mapper will resolve an `MvExpand(limit=n)` into a `LimitExec(n) + MvExpandExec`; after physical planning, even if that fragment remains the same, it will be re-translated to `Limit(n) + MvExpand` and serialized.

Reply:
I think I get it.

> To clarify, we don't need to serialize it now, but we could.

The fragment will contain the logical plan, the "original" one. So theoretically, we might need to serialise an `MvExpand` with a non-null limit. In which case the assertion would trip. But `MvExpand`'s limit would be non-`null` only if a limit was pushed down over it and since limit is a pipeline breaker, the plan will be latest broken down at the limit, so MvExpand will, in current form, never make it onto the nodes. Right?
The translation to physical and the mapping wouldn't then influence this, I guess.

Reply:
> and since limit is a pipeline breaker, the plan will be latest broken down at the limit, so MvExpand will, in current form, never make it onto the nodes. Right?

Correct, and it's enough to guarantee that limited MV_EXPAND will never make it into the data nodes.

> The translation to physical and the mapping wouldn't then influence this, I guess.

I think you're right